packages/grafana-ui/src/components/Forms/Input/Input.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { FC, HTMLProps, ReactNode } from 'react';
+import { GrafanaTheme } from '@grafana/data';
+import { css, cx } from 'emotion';
+import { getFocusStyle } from '../commonStyles';
+import { selectThemeVariant, stylesFactory, useTheme } from '../../../themes';
+import { Icon } from '../../Icon/Icon';
+import { useClientRect } from '../../../utils/useClientRect';
+
+export interface Props extends Omit<HTMLProps<HTMLInputElement>, 'prefix'> {
+  /** Show an invalid state around the input */
+  invalid?: boolean;
+  /** Show an icon as a prefix in the input */
+  prefix?: JSX.Element | string | null;
+  /** Show a loading indicator as a suffix in the input */
+  loading?: boolean;
+  /** Add a component as an addon before the input  */
+  addonBefore?: ReactNode;
+  /** Add a component as an addon after the input */
+  addonAfter?: ReactNode;
+}
+
+const getInputStyle = stylesFactory((theme: GrafanaTheme, invalid = false) => {
+  const colors = theme.colors;
+  const backgroundColor = selectThemeVariant({ light: colors.white, dark: colors.gray15 }, theme.type);
```

Comment:
Can we use theme variables introduced in https://github.com/grafana/grafana/pull/19874 to retrive colors and spacings?

Reply:
yup, there are variable introduced so that we can reuse same colors for i.e select ltr on